api/http_client.go

```diff
@@ -64,6 +68,8 @@ func NewHTTPClient(opts HTTPClientOptions) (*http.Client, error) {
 		client.Transport = AddAuthTokenHeader(client.Transport, opts.Config)
 	}
 
+	client.Transport = SanitizeASCIIControlCharacters(client.Transport)
```

Comment:
Does this middleware also affect the `gh api` command? Since `gh api` is meant for low-level, programmatic use, maybe it would be good if it remained unaffected.

Reply:
It does currently affect the `gh api` command. In my mind I think we should sanitize those requests as well as we want to protect our users throughout our tool. I can envision people copying and pasting `gh api` commands that request a URL with malicious data in the response. If there is a valid use case for leaving these requests un-sanitized then I am open to it but I have been unable to think of one.

Reply:
The way I see it, `gh api` is meant for scripts, not necessarily to implement presentation. But agreed that flags like `--jq` and especially `--template` can be used to format data for the terminal. In those cases it's good if ANSI escape codes are disabled. Maybe it's okay that this middleware sanitizes all those.